semgrep-core/src/parsing/tree_sitter/Parse_swift_tree_sitter.ml

```diff
@@ -2038,19 +2067,22 @@ and map_universally_allowed_pattern (env : env)
         | Some tok -> (* "case" *) token env tok
         | None -> todo env ()
       in
-      let v2 =
-        match v2 with
-        | Some x -> map_user_type env x
-        | None -> todo env ()
-      in
       let v3 = (* dot_custom *) token env v3 in
-      let v4 = map_bound_identifier env v4 in
-      let v5 =
-        match v5 with
-        | Some x -> map_tuple_pattern env x
-        | None -> todo env ()
+      let id = map_bound_identifier env v4 in
+      let id_info = G.empty_id_info () in
+      let pat_init = G.PatId (id, id_info) in
+      let add_pat_args name pat =
+        match Option.map (map_tuple_pattern env) v5 with
+        | None -> pat
+        | Some (G.PatTuple (_, pats, _)) -> G.PatConstructor (name, pats)
+        | _ -> raise Common.Impossible
```

Comment:
Might it be possible to refactor so that this isn't necessary? Maybe `map_tuple_pattern` can return just the contents of a `PatTuple`?